src/components/Proposal/Create/SchemeForms/TagsSelector.tsx

```diff
@@ -0,0 +1,168 @@
+    if (this.props.onChange) {
+      this.props.onChange(tags.map((tag: Tag) => tag.text));    
+    }
+  }
+
+  private handleDelete = () => (i: number): void => {
+    const tags = this.state.workingTags.filter((_tag: Tag, index: number) => index !== i);
+    this.setState({ workingTags: tags });
+    this.emitOnChange(tags);
+  }
+ 
+  private handleAddition = () => (tag: Tag): void => {
+    const tags = [...this.state.workingTags, tag];
+    this.setState({ workingTags:  tags });
+    this.emitOnChange(tags);
+  }
+
+  private handleDrag = () => (tag: Tag, currPos: number, newPos: number): void => {
+    const tags = this.state.workingTags.slice();
+
+    tags.splice(currPos, 1);
+    tags.splice(newPos, 0, tag);
+
+    this.setState({ workingTags: tags });
+    this.emitOnChange(tags);
+  }
+
+  /**
+   * return Map where key is tag, value is count of tags
+   * @param allTags 
+   */
+  private groupTags(allTags: Array<string>): Map<string,number> {
+    const map = new Map<string,number>();
+    allTags.forEach(tag => {
+      tag = tag.toLocaleLowerCase();
+      if (!map.has(tag)) {
+        map.set(tag,0);
+      }
+      map.set(tag,map.get(tag)+1);
+    });
+    return map;
+  }
+
+  public render(): RenderOutput {
+    const { workingTags } = this.state;
+    const { readOnly, darkTheme } = this.props;
+    const allTags: Array<Array<string>> = this.props.data;
+    let suggestions: Array<ITagEx>;
+    
+    if (allTags && allTags.length) {
+      suggestions = new Array<ITagEx>();
+      /**
+       * concatenate all the proposals' lists of tags, then group by frequency
+       */
+      const tagCounts = this.groupTags(allTags.reduce((acc: Array<string>, tags: Array<string>): Array<string> => {
+        if (tags && tags.length) {
+          acc = acc.concat(tags);
+        }
+        return acc;
+      }, new Array<string>())
+      );
+      // sort by descending tag count
+      const sortedTags = Array.from(tagCounts.keys()).sort((a: string, b: string): number => tagCounts.get(b) - tagCounts.get(a));
+      for (const tag of sortedTags) {
+        suggestions.push({ id: tag, text: tag, count: tagCounts.get(tag)});
+      }
+    }
+
+    return <div className={classNames({
+      [css.reactTagsContainer]: true,
+      ["darkTheme"]: darkTheme,
+    })}>
+      {
+        // from here: https://github.com/prakhar1989/react-tags
+      }
+      <ReactTags
+        tags={workingTags}
+        suggestions={suggestions}
+        handleDelete={this.handleDelete()}
+        handleAddition={this.handleAddition()}
+        handleDrag={this.handleDrag()}
+        delimiters={delimiters}
+        autocomplete={1}
+        readOnly={!!readOnly}
+        autofocus={false}
+        allowDragDrop={false} // because we have no way to persist the tab order
+        // eslint-disable-next-line react/jsx-no-bind
+        renderSuggestion = {( tag: ITagEx ) => <div>{tag.text} <span className="count">({tag.count})</span></div>}
```

Comment:
it is RED, mus it be red?

Reply:
It is the spec.  There is an open question about that for @A-Zak

Reply:
done